scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingMethodImpl.java

```diff
@@ -167,28 +168,38 @@ public int schedule() {
             List<JobDescriptor> descriptors = new ArrayList<>(jobMap.size());
             descriptors.addAll(jobMap.values());
 
+
+            //get rmState and update it in scheduling policy
+            RMState rmState = getRMProxiesManager().getRmProxy().getState();
+            currentPolicy.setRMState(rmState);
+            int freeResourcesNb = rmState.getFreeNodesNumber();
+            logger.debug("eligible nodes : " + freeResourcesNb);
+            //if there is no free resources, stop it right now
+            if (freeResourcesNb == 0) {
+                return numberOfTaskStarted;
+            }
+
+            Vector<EligibleTaskDescriptor> policyTasks = currentPolicy
+                    .getOrderedTasks(descriptors);
+
             //ask the policy all the tasks to be schedule according to the jobs list.
             //and filter them using internal policy
-            LinkedList<EligibleTaskDescriptor> taskRetrievedFromPolicy = internalPolicy.filter(currentPolicy
-                    .getOrderedTasks(descriptors));
+
+            LinkedList<EligibleTaskDescriptor> taskRetrievedFromPolicy = new LinkedList<>(policyTasks);
```

Comment:
Why not changing the signature of Policy#getOrderedTasks to return a linked list? It will prevent to create a copy. Using a non synchronized linked list seems ok since the list is not used by multiple threads.